internal/pkg/term/syncbuffer/termprinter.go

```diff
@@ -0,0 +1,184 @@
+
+// NewLabeledTermPrinter returns a LabeledTermPrinter that can print to the terminal filewriter from buffers.
+func NewLabeledTermPrinter(fw FileWriter, bufs []*LabeledSyncBuffer, opts ...LabeledTermPrinterOption) (*LabeledTermPrinter, error) {
+	width, err := terminalWidth()
+	if err != nil {
+		return nil, fmt.Errorf("get terminal width: %w", err)
+	}
+
+	ltp := &LabeledTermPrinter{
+		term:      fw,
+		buffers:   bufs,
+		termWidth: width,
+		numLines:  -1, // By default set numlines to -1 to print all from buffers.
+	}
+	for _, opt := range opts {
+		opt(ltp)
+	}
+	return ltp, nil
+}
+
+// WithNumLines sets the numlines of LabeledTermPrinter.
+func WithNumLines(n int) LabeledTermPrinterOption {
+	return func(ltp *LabeledTermPrinter) {
+		ltp.numLines = n
+	}
+}
+
+// WithPadding sets the padding of LabeledTermPrinter.
+func WithPadding(n int) LabeledTermPrinterOption {
+	return func(ltp *LabeledTermPrinter) {
+		ltp.padding = n
+	}
+}
+
+// isDone returns true if all the buffers are done else returns false.
+func (ltp *LabeledTermPrinter) isDone() bool {
+	for _, buf := range ltp.buffers {
+		if !buf.syncBuf.IsDone() {
+			return false
+		}
+	}
+	return true
+}
+
+// Print prints the label and the last N lines of logs from each buffer to the termPrinter fileWriter.
+// If numLines is -1 then print all the values from buffers.
+// It polls each buffer until all the buffers are marked done,
+// and erases the previous output after sleeping for a short duration.
+func (ltp *LabeledTermPrinter) Print() {
+	if ltp.numLines == -1 {
+		ltp.PrintAll()
+		return
+	}
+	for {
+		ltp.prevWrittenLines = 0
+		for _, buf := range ltp.buffers {
+			logs := buf.syncBuf.strings()
+			if len(logs) == 0 {
+				continue
+			}
+			outputLogs := ltp.lastNLines(logs)
+			ltp.writeLines(buf.label, outputLogs)
+			ltp.prevWrittenLines += ltp.calculateLinesCount(append(outputLogs, buf.label))
+		}
+		if ltp.isDone() {
+			break
+		}
+		time.Sleep(pollIntervalForSyncBuffer)
+		cursor.EraseLinesAbove(os.Stderr, ltp.prevWrittenLines)
+	}
+}
+
+// lastNLines returns the last N lines of the given logs where N is the value of tp.numLines.
+// If the logs slice contains fewer than N lines, all lines are returned.
+// If the given input logs are empty then return slice of empty strings.
+func (ltp *LabeledTermPrinter) lastNLines(logs []string) []string {
+	var start int
+	if len(logs) > ltp.numLines {
+		start = len(logs) - ltp.numLines
+	}
+	end := len(logs)
+
+	// Extract the last N lines
+	logLines := make([]string, ltp.numLines)
+	idx := 0
+	for start < end {
+		logLines[idx] = strings.TrimSpace(logs[start])
+		start++
+		idx++
+	}
+	return logLines
+}
+
+// writeLines writes a label and output logs to the terminal associated with the TermPrinter.
+func (ltp *LabeledTermPrinter) writeLines(label string, outputLogs []string) {
+	padding := ""
+	if ltp.padding != 0 {
+		padding = strings.Repeat(" ", ltp.padding)
+	}
```

Comment:
this should be safe even when 0
```suggestion
	padding := strings.Repeat(" ", ltp.padding)
```